src/dodal/devices/bimorph_mirrors/CAENels_bimorph_mirror_interface.py

```diff
@@ -0,0 +1,273 @@
+        """Takes an attribute and returns a list of Signals that share that attribute across all channels.
+
+        Args:
+            channel_attribute: A ChannelAttribute enum representing the grouping of signals to be returned.
+                eg: ChannelAttribute.VOUT would yield a list of objects representing [C1:VOUT, C2:VOUT, ..., CX:VOUT]
+                for an X-channel bimorph
+
+        Returns:
+            A list containing X Signals where X is the number of channels of the bimorph.
+        """
+        if channel_attribute == ChannelAttribute.VTRGT:
+            return self._voltage_target_channels
+
+        elif channel_attribute == ChannelAttribute.VTRGT_RBV:
+            return self._voltage_target_readback_value_channels
+
+        elif channel_attribute == ChannelAttribute.SHIFT:
+            return self._shift_channels
+
+        elif channel_attribute == ChannelAttribute.VOUT:
+            return self._voltage_out_channels
+
+        elif channel_attribute == ChannelAttribute.VOUT_RBV:
+            return self._voltage_out_readback_value_channels
+
+        elif channel_attribute == ChannelAttribute.STATUS:
+            return self._status_channels
+
+    def read_from_all_channels_by_attribute(
+        self, channel_attribute: ChannelAttribute
+    ) -> list:
+        """Takes an attribute and returns a list of values read from all signals sharing that attribute.
+
+        Args:
+            channel_attribute: A ChannelAttribute enum representing the grouping of signals to be returned.
+                eg: ChannelAttribute.VOUT would yield a list of objects representing [C1:VOUT, C2:VOUT, ..., CX:VOUT]
+                for an X-channel bimorph            channel_attribute: attribute grouping
+
+
+        Returns:
+            A list of values read from X signals related to the given attribute, where X is the numebr of channels in the bimorph.
+        """
+        channels = self.get_channels_by_attribute(channel_attribute)
+
+        return [self.parsed_protected_read(channel) for channel in channels]
+
+    def write_to_all_channels_by_attribute(
+        self, channel_attribute: ChannelAttribute, values: list
+    ) -> SubscriptionStatus:
+        """Writes given values to signals grouped by given attribute.
+
+        Args:
+            channel_attribute: A ChannelAttribute enum representing the grouping of signals to be written to.
+            values: A list of values of length equal to the number of channels of the bimorph
+
+        Returns:
+            A SubscriptionStatus object tracking completion of operations
+        """
+        channels = self.get_channels_by_attribute(channel_attribute)
+
+        status = StatusBase()
+        status.set_finished()
+
+        for i, (channel, value) in enumerate(zip(channels, values)):
+            status &= self.protected_set(channel, value)
+
+            if i >= len(channels) - 1:
```

Comment:
Does this mean we always return without waiting? Can we ever reach the other return statement?